File: DNN Platform/Admin Modules/Dnn.Modules.ModuleCreator/Templates/C#/Scheduled Job/templateJob.cs
// 
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the MIT License. See LICENSE file in the project root for full license information.
// 
using System;
using DotNetNuke;

namespace _OWNER_._MODULE_
{

    public class _MODULE_Job : DotNetNuke.Services.Scheduling.SchedulerClient
    {

        public _MODULE_Job(DotNetNuke.Services.Scheduling.ScheduleHistoryItem objScheduleHistoryItem) : base()
        {
            this.ScheduleHistoryItem = objScheduleHistoryItem;
        }

        public override void DoWork()
        {
            try {
                this.Progressing();
                string strMessage = Processing();
                this.ScheduleHistoryItem.Succeeded = true;
                this.ScheduleHistoryItem.AddLogNote("_MODULE_ Succeeded");
            } catch (Exception exc) {
                this.ScheduleHistoryItem.Succeeded = false;
                this.ScheduleHistoryItem.AddLogNote("_MODULE_ Failed");
                this.Errored(ref exc);
            }
        }

        public string Processing()
        {
            string Message = "";
            return Message;
        }

    }

}

File: DNN Platform/Admin Modules/Dnn.Modules.ModuleCreator/Templates/VB/Scheduled Job/templateJob.vb
#Region "Copyright"

' 
' Copyright (c) _YEAR_
' by _OWNER_
' 

#End Region

Imports System
Imports DotNetNuke

Namespace _OWNER_._MODULE_

    Public Class _MODULE_Job

        Inherits DotNetNuke.Services.Scheduling.SchedulerClient

        Public Sub New(ByVal objScheduleHistoryItem As DotNetNuke.Services.Scheduling.ScheduleHistoryItem)
            MyBase.New()
            Me.ScheduleHistoryItem = objScheduleHistoryItem
        End Sub

        Public Overrides Sub DoWork()
            Try
                Me.Progressing()
                Dim strMessage As String = Processing()
                Me.ScheduleHistoryItem.Succeeded = True
                Me.ScheduleHistoryItem.AddLogNote("_MODULE_ Succeeded")
            Catch exc As Exception    
                Me.ScheduleHistoryItem.Succeeded = False
                Me.ScheduleHistoryItem.AddLogNote("_MODULE_ Failed")
                Me.Errored(exc)
            End Try
        End Sub

        Public Function Processing() As String
            Dim Message As String = ""
            Return Message 
        End Function

    End Class

End Namespace

